docs/guide/building-an-advanced-scene.md

```diff
@@ -4,238 +4,456 @@ type: guide
+First, let us whitelist the entities that the cursor's raycaster is checking
+for intersections against. That way, the cursor will only click if something
+can be clicked, and it is also better for performance. The cursor is built on
+top of the [raycaster component][raycaster], and we can configure the
+raycaster. We update the raycaster component's `objects` property which takes a
+selector:
+
+```html
+<a-cursor id="cursor" raycaster="objects: .link">
+```
+
+This list will be populated once the raycaster component attaches.
+Unfortunately since the links are templated, they won't be found at that time.
+What we can do is write a component that refreshes our raycaster when the link
+attaches. Here will be the skeleton of our component:
+
+```
+AFRAME.registerComponent('update-raycaster', {
+  schema: {
+    // ...
+  },
 
-```js
-AFRAME.registerComponent('click-listener', {
   init: function () {
-    var el = this.el;
-    window.addEventListener('click', function () {
-      el.emit('click', null, false);
-    });
+    // ...
   }
 });
 ```
 
-From HTML, we define the laser mixin and attach the spawner and click-listener components to the player. When we click, the spawner component will generate a laser starting in front of the camera:
+First, we fill out the [`schema`][schema] so that we can pass in which raycaster to
+update. We make it a single-property schema that takes a selector such that we can
+simply do `update-raycaster="#cursor"`:
 
-<p data-height="500" data-theme-id="0" data-slug-hash="jqEjvB" data-default-tab="html" data-user="mozvr" class="codepen">See the Pen <a href="http://codepen.io/team/mozvr/pen/jqEjvB/">Laser Shooter - Step 4</a> by MozVR (<a href="http://codepen.io/team/mozvr">@mozvr</a>) on <a href="http://codepen.io">CodePen</a>.</p>
+```js
+AFRAME.registerComponent('update-raycaster', {
+  schema: {
+    type: 'selector'
+  },
 
-### projectile Component
+  init: function () {
+    // ...
+  }
+});
+```
 
-Now lasers will spawn in front of us when we click, but we need them to fire and travel. In the spawner component, we had the laser point in the rotation of the camera, and we rotated it 90-degrees around the X-axis to align it correctly. We can add a projectile component to have the laser travel straight in the direction it's already facing (its local Y-axis in this case):
+Then we use that data to actually update the raycaster on the `init` lifecycle
+method, which is called when the component is attached to the entity. We grab
+the raycaster and update it:
 
 ```js
-AFRAME.registerComponent('projectile', {
+AFRAME.registerComponent('update-raycaster', {
   schema: {
-    speed: { default: -0.4 }
+    type: 'selector'
   },
 
-  tick: function () {
-    this.el.object3D.translateY(this.data.speed);
+  init: function () {
+    var raycasterEl = this.data;
+    this.data.components.raycaster.refreshObjects();
   }
 });
 ```
 
-Then attach the projectile component to the laser mixin:
+### Set Image Component
```

Comment:
you [hyphenate](https://github.com/aframevr/aframe/pull/1494/files#diff-152a0cc3c6c095bc8f95916fa320645dR348) this below. should probably do the same here.